File: src/main/java/net/hectus/neobb/turn/person_game/categorization/DefensiveCategory.java
package net.hectus.neobb.turn.person_game.categorization;

import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;
import org.bukkit.Material;

public interface DefensiveCategory extends Category {
    @Override
    default TextColor categoryColor() {
        return NamedTextColor.GREEN;
    }

    @Override
    default Material categoryItem() {
        return Material.RED_STAINED_GLASS;
    }

    @Override
    default String categoryName() {
        return "defensive";
    }

    // ========== CUSTOM CATEGORY LOGIC ==========

    void applyDefense();
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/categorization/DefensiveCategory.kt
package net.hectus.neobb.turn.person_game.categorization

import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Material

interface DefensiveCategory : Category {
    override val categoryColor: TextColor get() = NamedTextColor.GREEN

    override val categoryItem: Material get() = Material.RED_STAINED_GLASS
    override val categoryName: String get() = "defensive"

    // ========== CUSTOM CATEGORY LOGIC ==========
    fun applyDefense()
}
